windows-installer/construct.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+import ctypes
+import subprocess
+import os
+import sys
+import shutil
+
+
+def main():
+    print('Gather runtime data')
+
+    try:
+        subprocess.check_output(['choco', '--version'])
+    except subprocess.CalledProcessError:
+        raise RuntimeError('Error: Chocolatey (https://chocolatey.org/) needs'
+                           'to be installed to run this script.')
+
+    script_path = os.path.realpath(__file__)
+    repo_path = os.path.dirname(os.path.dirname(script_path))
+    build_path = os.path.join(repo_path, 'windows-installer', 'build')
+
+    venv_path = os.path.join(build_path, 'venv-config')
+    venv_python = os.path.join(venv_path, 'Scripts', 'python.exe')
+    installer_cfg_path = os.path.join(build_path, 'installer.cfg')
+    wheels_path = os.path.join(build_path, 'wheels')
+
+    certbot_version = subprocess.check_output([sys.executable, '-c', 'import certbot; print(certbot.__version__)'],
+                                              universal_newlines=True, cwd=repo_path).strip()
+
+    certbot_packages = ['acme', '.']
+    certbot_packages.extend([name for name in os.listdir(repo_path) if name.startswith('certbot-dns-')])
+
+    print('Copy assets')
+
+    os.makedirs(build_path, exist_ok=True)
+    shutil.copy(os.path.join(repo_path, 'windows-installer', 'certbot.ico'), build_path)
+    shutil.copy(os.path.join(repo_path, 'windows-installer', 'run.py'), build_path)
+
+    print('Prepare pynsist config')
+
+    with open(os.path.join(installer_cfg_path), 'w') as file_h:
+        file_h.write("""\
+[Application]
+name=Certbot
+version={certbot_version}
+icon=certbot.ico
+publisher=Electronic Frontier Fundation
+script=run.py
+
+[Build]
+directory=nsis
+installer_name=certbot-{certbot_version}-win32_install.exe
```

Comment:
You can ignore these suggestions if you want but:

1. I didn't like the one underscore in a sea of hyphens.
2. Giving this file name the noun "installer" instead of the verb "install" seems better to me with my extremely outdated knowledge of Windows conventions.
```suggestion
installer_name=certbot-{certbot_version}-win32-installer.exe
```

Reply:
I changed to `installer`. I keep a underscore for `win_amd64`, since it is the convention for 64 bits Windows wheels, and used in some other significant situations, like in executables built with go for instance.